# Dia 1 - Importação de dados

In [56]:
import pandas as pd

## Coletando dados do GitHub

In [57]:
dados_2010_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20101.csv?raw=true')
dados_2010_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20102.csv?raw=true')
dados_2011_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20111.csv?raw=true')
dados_2011_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20112.csv?raw=true')
dados_2012_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20121.csv?raw=true')
dados_2012_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20122.csv?raw=true')
dados_2013_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20131.csv?raw=true')
dados_2013_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20132.csv?raw=true')
dados_2014_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20141.csv?raw=true')
dados_2014_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20142.csv?raw=true')
dados_2015_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20151.csv?raw=true')
dados_2015_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20152.csv?raw=true')
dados_2016_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20161.csv?raw=true')
dados_2016_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20162.csv?raw=true')
dados_2017_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20171.csv?raw=true')
dados_2017_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20172.csv?raw=true')
dados_2018_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20181.csv?raw=true')
dados_2018_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20182.csv?raw=true')
dados_2019_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20191.csv?raw=true')
dados_2019_2 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20192.csv?raw=true')
dados_2020_1 = pd.read_csv('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/blob/main/Dia_1-Importando_dados/Datasets/dados_emprestimos/emprestimos-20201.csv?raw=true')

In [58]:
exemplares = pd.read_parquet ('https://github.com/FranciscoFoz/7_Days_of_Code_Alura-Python-Pandas/raw/main/Dia_1-Importando_dados/Datasets/dados_exemplares.parquet')

## Concatenando DataFrames

In [59]:
emprestimos = pd.concat([dados_2010_1, dados_2010_2, dados_2011_1, dados_2011_2, dados_2012_1, dados_2012_2, dados_2013_1, dados_2013_2, dados_2014_1, dados_2014_2, dados_2015_1, dados_2015_2, dados_2016_1, dados_2016_2, dados_2017_1, dados_2017_2, dados_2018_1, dados_2018_2, dados_2019_1, dados_2019_2, dados_2020_1])
emprestimos

id_emprestimo codigo_barras                 data_renovacao  \
0             709684       L095049                            NaN   
1             709685       L167050                            NaN   
2             709686    2006017618  2010/01/26 08:07:01.738000000   
3             709687       L184117  2010/01/18 11:07:46.470000000   
4             709684       L095049                            NaN   
...              ...           ...                            ...   
26556        2989086    2008047044                            NaN   
26557        2989087    2008047047                            NaN   
26558        2989088    2010052905                            NaN   
26559        2989089    2010048023                            NaN   
26560        2989090    2013106838                            NaN   

                     data_emprestimo                 data_devolucao  \
0      2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
1      2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
2      2010/01/04 08:08:44.081000000  2010/02/25 07:36:25.800000000   
3      2010/01/04 08:24:21.284000000  2010/02/03 08:58:45.692000000   
4      2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
...                              ...                            ...   
26556  2020/05/05 19:51:06.059000000  2021/05/13 14:53:31.000000000   
26557  2020/05/05 19:51:06.135000000  2021/05/13 14:53:38.000000000   
26558  2020/05/05 19:51:06.164000000  2021/05/13 15:01:46.000000000   
26559  2020/06/09 11:54:02.669000000  2021/07/21 15:33:55.000000000   
26560  2020/06/09 11:54:02.852000000  2021/07/21 15:33:45.000000000   

       matricula_ou_siape             tipo_vinculo_usuario  
0            2.008023e+09               ALUNO DE GRADUAÇÃO  
1            2.008023e+09               ALUNO DE GRADUAÇÃO  
2            2.008112e+09           ALUNO DE PÓS-GRADUAÇÃO  
3            2.007211e+08               ALUNO DE GRADUAÇÃO  
4            2.008023e+09               ALUNO DE GRADUAÇÃO  
...                   ...                              ...  
26556        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
26557        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
26558        3.067431e+06  SERVIDOR TÉCNICO-ADMINISTRATIVO  
26559        2.020101e+10           ALUNO DE PÓS-GRADUAÇÃO  
26560        2.020101e+10           ALUNO DE PÓS-GRADUAÇÃO  

[2258018 rows x 7 columns]

## Mesclando dados de emprestimos com dados dos exemplares

In [60]:
emprestimos_completo = emprestimos.merge(exemplares)
emprestimos_completo

id_emprestimo codigo_barras                 data_renovacao  \
0               709684       L095049                            NaN   
1               709684       L095049                            NaN   
2               726965       L095049  2010/03/31 12:41:08.684000000   
3               709685       L167050                            NaN   
4               709685       L167050                            NaN   
...                ...           ...                            ...   
2072553        2989018    2009036492                            NaN   
2072554        2989020    2014062640                            NaN   
2072555        2989032    2018003324                            NaN   
2072556        2989049    2006024313                            NaN   
2072557        2989061    2018031352                            NaN   

                       data_emprestimo                 data_devolucao  \
0        2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
1        2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
2        2010/03/01 12:40:56.212000000  2010/04/30 14:42:35.366000000   
3        2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
4        2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
...                                ...                            ...   
2072553  2020/03/17 19:49:10.187000000  2020/12/03 08:12:20.000000000   
2072554  2020/03/17 19:51:15.700000000  2021/01/07 11:16:28.000000000   
2072555  2020/03/17 20:00:11.416000000  2021/08/30 13:29:44.000000000   
2072556  2020/03/17 20:47:18.741000000                            NaN   
2072557  2020/03/17 21:22:22.138000000  2021/07/26 15:05:20.000000000   

         matricula_ou_siape    tipo_vinculo_usuario  id_exemplar  \
0              2.008023e+09      ALUNO DE GRADUAÇÃO        13259   
1              2.008023e+09      ALUNO DE GRADUAÇÃO        13259   
2              2.009122e+09  ALUNO DE PÓS-GRADUAÇÃO        13259   
3              2.008023e+09      ALUNO DE GRADUAÇÃO        70865   
4              2.008023e+09      ALUNO DE GRADUAÇÃO        70865   
...                     ...                     ...          ...   
2072553        2.015011e+10      ALUNO DE GRADUAÇÃO       274820   
2072554        2.019004e+10      ALUNO DE GRADUAÇÃO      1157915   
2072555        2.019102e+10  ALUNO DE PÓS-GRADUAÇÃO      1317173   
2072556        2.015013e+10      ALUNO DE GRADUAÇÃO       206073   
2072557        2.020001e+10      ALUNO DE GRADUAÇÃO      1342535   

                   colecao                                         biblioteca  \
0        Acervo Circulante                     Biblioteca Central Zila Mamede   
1        Acervo Circulante                     Biblioteca Central Zila Mamede   
2        Acervo Circulante                     Biblioteca Central Zila Mamede   
3        Acervo Circulante                     Biblioteca Central Zila Mamede   
4        Acervo Circulante                     Biblioteca Central Zila Mamede   
...                    ...                                                ...   
2072553  Acervo Circulante                     Biblioteca Central Zila Mamede   
2072554  Acervo Circulante                     Biblioteca Central Zila Mamede   
2072555  Acervo Circulante  Biblioteca Setorial da Faculdade de Ciências d...   
2072556  Acervo Circulante                     Biblioteca Central Zila Mamede   
2072557  Acervo Circulante                     Biblioteca Central Zila Mamede   

        status_material  localizacao  registro_sistema  
0               REGULAR          630              4225  
1               REGULAR          630              4225  
2               REGULAR          630              4225  
3               REGULAR          647             25009  
4               REGULAR          647             25009  
...                 ...          ...               ...  
2072553         REGULAR          676            120845  
2072554         REGULAR          

## Verificando duplicatas

In [61]:
emprestimos_completo.value_counts()

id_emprestimo  codigo_barras  data_renovacao                 data_emprestimo                data_devolucao                 matricula_ou_siape  tipo_vinculo_usuario             id_exemplar  colecao            biblioteca                                                                                status_material  localizacao  registro_sistema
709704         2006021442     2010/01/17 18:56:43.923000000  2010/01/04 09:58:46.522000000  2010/02/01 13:59:02.528000000  2.008010e+09        ALUNO DE GRADUAÇÃO               199419       Acervo Circulante  Biblioteca Central Zila Mamede                                                            REGULAR          699          77866               7
709705         2009032168     2010/01/17 18:56:43.909000000  2010/01/04 09:58:46.549000000  2010/01/20 13:14:34.464000000  2.008010e+09        ALUNO DE GRADUAÇÃO               273487       Acervo Circulante  Biblioteca Central Zila Mamede                                                            REGULAR

## Removendo duplicatas

In [62]:
emprestimos_completo = emprestimos_completo.drop_duplicates()
emprestimos_completo

id_emprestimo codigo_barras                 data_renovacao  \
0               709684       L095049                            NaN   
2               726965       L095049  2010/03/31 12:41:08.684000000   
3               709685       L167050                            NaN   
5               717763       L167050                            NaN   
6               729996       L167050  2010/03/17 09:08:26.874000000   
...                ...           ...                            ...   
2072553        2989018    2009036492                            NaN   
2072554        2989020    2014062640                            NaN   
2072555        2989032    2018003324                            NaN   
2072556        2989049    2006024313                            NaN   
2072557        2989061    2018031352                            NaN   

                       data_emprestimo                 data_devolucao  \
0        2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
2        2010/03/01 12:40:56.212000000  2010/04/30 14:42:35.366000000   
3        2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
5        2010/02/11 11:16:30.335000000  2010/03/04 08:54:32.262000000   
6        2010/03/04 13:36:26.599000000  2010/03/31 18:04:29.669000000   
...                                ...                            ...   
2072553  2020/03/17 19:49:10.187000000  2020/12/03 08:12:20.000000000   
2072554  2020/03/17 19:51:15.700000000  2021/01/07 11:16:28.000000000   
2072555  2020/03/17 20:00:11.416000000  2021/08/30 13:29:44.000000000   
2072556  2020/03/17 20:47:18.741000000                            NaN   
2072557  2020/03/17 21:22:22.138000000  2021/07/26 15:05:20.000000000   

         matricula_ou_siape    tipo_vinculo_usuario  id_exemplar  \
0              2.008023e+09      ALUNO DE GRADUAÇÃO        13259   
2              2.009122e+09  ALUNO DE PÓS-GRADUAÇÃO        13259   
3              2.008023e+09      ALUNO DE GRADUAÇÃO        70865   
5              2.008050e+09      ALUNO DE GRADUAÇÃO        70865   
6              2.007226e+08      ALUNO DE GRADUAÇÃO        70865   
...                     ...                     ...          ...   
2072553        2.015011e+10      ALUNO DE GRADUAÇÃO       274820   
2072554        2.019004e+10      ALUNO DE GRADUAÇÃO      1157915   
2072555        2.019102e+10  ALUNO DE PÓS-GRADUAÇÃO      1317173   
2072556        2.015013e+10      ALUNO DE GRADUAÇÃO       206073   
2072557        2.020001e+10      ALUNO DE GRADUAÇÃO      1342535   

                   colecao                                         biblioteca  \
0        Acervo Circulante                     Biblioteca Central Zila Mamede   
2        Acervo Circulante                     Biblioteca Central Zila Mamede   
3        Acervo Circulante                     Biblioteca Central Zila Mamede   
5        Acervo Circulante                     Biblioteca Central Zila Mamede   
6        Acervo Circulante                     Biblioteca Central Zila Mamede   
...                    ...                                                ...   
2072553  Acervo Circulante                     Biblioteca Central Zila Mamede   
2072554  Acervo Circulante                     Biblioteca Central Zila Mamede   
2072555  Acervo Circulante  Biblioteca Setorial da Faculdade de Ciências d...   
2072556  Acervo Circulante                     Biblioteca Central Zila Mamede   
2072557  Acervo Circulante                     Biblioteca Central Zila Mamede   

        status_material  localizacao  registro_sistema  
0               REGULAR          630              4225  
2               REGULAR          630              4225  
3               REGULAR          647             25009  
5               REGULAR          647             25009  
6               REGULAR          647             25009  
...                 ...          ...               ...  
2072553         REGULAR          676            120845  
2072554         REGULAR          

# Dia 2 - Limpeza de dados

## Criando uma nova coluna com os valores da localização, para refletir a respectiva classe geral na CDU.


000 a 099: Generalidades. Ciência e conhecimento.

100 a 199: Filosofia e psicologia.

200 a 299: Religião.

300 a 399: Ciências sociais.

400 a 499: Classe vaga. Provisoriamente não ocupada.

500 a 599: Matemática e ciências naturais.

600 a 699: Ciências aplicadas.

700 a 799: Belas artes.

800 a 899: Linguagem. Língua. Linguística.

900 a 999: Geografia. Biografia. História."

In [63]:
CDU_lista = []
for CDU in emprestimos_completo['localizacao']:
  if(CDU < 100):
    CDU_lista.append('Generalidades')
  elif(CDU < 200):
    CDU_lista.append('Filosofia e psicologia')
  elif(CDU < 300):
    CDU_lista.append('Religião')
  elif(CDU < 400):
    CDU_lista.append('Ciências sociais')
  elif(CDU < 500):
    CDU_lista.append('Classe vaga')
  elif(CDU < 600):
    CDU_lista.append('Matemática e ciências naturais')
  elif(CDU < 700):
    CDU_lista.append('Ciências aplicadas')
  elif(CDU < 800):
    CDU_lista.append('Belas artes')
  elif(CDU < 900):
    CDU_lista.append('Linguagem')
  else:
    CDU_lista.append('Geografia. Biografia. História.')

emprestimos_completo['CDU'] = CDU_lista
emprestimos_completo.head(5)

<ipython-input-63-d910c1fe067f>:24: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  emprestimos_completo['CDU'] = CDU_lista


id_emprestimo codigo_barras                 data_renovacao  \
0         709684       L095049                            NaN   
2         726965       L095049  2010/03/31 12:41:08.684000000   
3         709685       L167050                            NaN   
5         717763       L167050                            NaN   
6         729996       L167050  2010/03/17 09:08:26.874000000   

                 data_emprestimo                 data_devolucao  \
0  2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
2  2010/03/01 12:40:56.212000000  2010/04/30 14:42:35.366000000   
3  2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
5  2010/02/11 11:16:30.335000000  2010/03/04 08:54:32.262000000   
6  2010/03/04 13:36:26.599000000  2010/03/31 18:04:29.669000000   

   matricula_ou_siape    tipo_vinculo_usuario  id_exemplar            colecao  \
0        2.008023e+09      ALUNO DE GRADUAÇÃO        13259  Acervo Circulante   
2        2.009122e+09  ALUNO DE PÓS-GRADUAÇÃO        13259  Acervo Circulante   
3        2.008023e+09      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   
5        2.008050e+09      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   
6        2.007226e+08      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   

                       biblioteca status_material  localizacao  \
0  Biblioteca Central Zila Mamede         REGULAR          630   
2  Biblioteca Central Zila Mamede         REGULAR          630   
3  Biblioteca Central Zila Mamede         REGULAR          647   
5  Biblioteca Central Zila Mamede         REGULAR          647   
6  Biblioteca Central Zila Mamede         REGULAR          647   

   registro_sistema                 CDU  
0              4225  Ciências aplicadas  
2              4225  Ciências aplicadas  
3             25009  Ciências aplicadas  
5             25009  Ciências aplicadas  
6             25009  Ciências aplicadas

Conferindo os dados

In [64]:
((emprestimos_completo['localizacao'] >= 300) & (emprestimos_completo['localizacao'] < 400)).sum()

369536

In [65]:
emprestimos_completo['CDU'].value_counts()

CDU
Ciências aplicadas                 1425473
Ciências sociais                    369536
Matemática e ciências naturais       68744
Generalidades                        62521
Religião                             62295
Filosofia e psicologia               60563
Geografia. Biografia. História.       7989
Belas artes                           7911
Linguagem                             7490
Name: count, dtype: int64

## Excluindo coluna registro_sistema

In [66]:
emprestimos_completo = emprestimos_completo.drop(columns = ['registro_sistema'])
emprestimos_completo.head(5)

id_emprestimo codigo_barras                 data_renovacao  \
0         709684       L095049                            NaN   
2         726965       L095049  2010/03/31 12:41:08.684000000   
3         709685       L167050                            NaN   
5         717763       L167050                            NaN   
6         729996       L167050  2010/03/17 09:08:26.874000000   

                 data_emprestimo                 data_devolucao  \
0  2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
2  2010/03/01 12:40:56.212000000  2010/04/30 14:42:35.366000000   
3  2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
5  2010/02/11 11:16:30.335000000  2010/03/04 08:54:32.262000000   
6  2010/03/04 13:36:26.599000000  2010/03/31 18:04:29.669000000   

   matricula_ou_siape    tipo_vinculo_usuario  id_exemplar            colecao  \
0        2.008023e+09      ALUNO DE GRADUAÇÃO        13259  Acervo Circulante   
2        2.009122e+09  ALUNO DE PÓS-GRADUAÇÃO        13259  Acervo Circulante   
3        2.008023e+09      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   
5        2.008050e+09      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   
6        2.007226e+08      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   

                       biblioteca status_material  localizacao  \
0  Biblioteca Central Zila Mamede         REGULAR          630   
2  Biblioteca Central Zila Mamede         REGULAR          630   
3  Biblioteca Central Zila Mamede         REGULAR          647   
5  Biblioteca Central Zila Mamede         REGULAR          647   
6  Biblioteca Central Zila Mamede         REGULAR          647   

                  CDU  
0  Ciências aplicadas  
2  Ciências aplicadas  
3  Ciências aplicadas  
5  Ciências aplicadas  
6  Ciências aplicadas

## Verificando tipo de dados

In [67]:
emprestimos_completo.dtypes

id_emprestimo             int64
codigo_barras            object
data_renovacao           object
data_emprestimo          object
data_devolucao           object
matricula_ou_siape      float64
tipo_vinculo_usuario     object
id_exemplar               int64
colecao                  object
biblioteca               object
status_material          object
localizacao               int64
CDU                      object
dtype: object

## Transformando coluna matricula_ou_siape em STRING

In [68]:
emprestimos_completo['matricula_ou_siape'] = emprestimos_completo['matricula_ou_siape'].astype(str)

### Checando

In [69]:
emprestimos_completo.dtypes

id_emprestimo            int64
codigo_barras           object
data_renovacao          object
data_emprestimo         object
data_devolucao          object
matricula_ou_siape      object
tipo_vinculo_usuario    object
id_exemplar              int64
colecao                 object
biblioteca              object
status_material         object
localizacao              int64
CDU                     object
dtype: object

In [71]:
emprestimos_completo.head(5)

id_emprestimo codigo_barras                 data_renovacao  \
0         709684       L095049                            NaN   
2         726965       L095049  2010/03/31 12:41:08.684000000   
3         709685       L167050                            NaN   
5         717763       L167050                            NaN   
6         729996       L167050  2010/03/17 09:08:26.874000000   

                 data_emprestimo                 data_devolucao  \
0  2010/01/04 07:44:10.721000000  2010/01/05 16:26:12.662000000   
2  2010/03/01 12:40:56.212000000  2010/04/30 14:42:35.366000000   
3  2010/01/04 07:44:10.750000000  2010/01/12 07:34:13.934000000   
5  2010/02/11 11:16:30.335000000  2010/03/04 08:54:32.262000000   
6  2010/03/04 13:36:26.599000000  2010/03/31 18:04:29.669000000   

  matricula_ou_siape    tipo_vinculo_usuario  id_exemplar            colecao  \
0       2008023265.0      ALUNO DE GRADUAÇÃO        13259  Acervo Circulante   
2       2009121584.0  ALUNO DE PÓS-GRADUAÇÃO        13259  Acervo Circulante   
3       2008023265.0      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   
5       2008050352.0      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   
6        200722565.0      ALUNO DE GRADUAÇÃO        70865  Acervo Circulante   

                       biblioteca status_material  localizacao  \
0  Biblioteca Central Zila Mamede         REGULAR          630   
2  Biblioteca Central Zila Mamede         REGULAR          630   
3  Biblioteca Central Zila Mamede         REGULAR          647   
5  Biblioteca Central Zila Mamede         REGULAR          647   
6  Biblioteca Central Zila Mamede         REGULAR          647   

                  CDU  
0  Ciências aplicadas  
2  Ciências aplicadas  
3  Ciências aplicadas  
5  Ciências aplicadas  
6  Ciências aplicadas